src/simulators.jl

```diff
@@ -841,3 +842,80 @@ function random_unit_vector(float_type, dims)
     vec = randn(float_type, dims)
     return vec / norm(vec)
 end
+
+
+"""
+    NoseHoover(; <keyword arguments>)
+
+    An NVT simulator that extends the traditional velocity verlet to also control the temperature of
+    the system. Finds the "version of dynamics" that is as close to the Hamiltonian as possible while maintaining temp.
+
+# Arguments
+- `dt::T`: the time step of the simulation.
+- `T_desired::K`: The temeprature targeted by the NoseHoover thermostat
+- `coupling::C=NoCoupling()`: the coupling which applies during the simulation.
+- `tau_damp::TD`: The temperature damping timescale, typically 100*dt
+- `zeta::Z`: Amount of damping requied to bring system to T_desired, based on damping timescale tau_damp.
+- `remove_CM_motion::Bool=true`: whether to remove the center of mass motion
+    every time step.
+"""
+mutable struct NoseHoover{DT, K, C, TD}
+    dt::DT
+    T_desired::K
```

Comment:
For consistency with other simulators I would go with `temperature` rather than `T_desired`. `damping` may also be clearer than `tau_damp`.